### PR TITLE
Support agency clearlist for field trip requests

### PR DESCRIPTION
**Pull request by miles-grant-ibigroup**
**Description:**
Field trip may not apply for some agencies. Makes use of OTP `unpreferred` to deprioritize these other agencies. Includes a conversion from Clear-list to banned list for convenience 

Blocked by https://github.com/opentripplanner/otp-ui/pull/746, https://github.com/opentripplanner/otp-react-redux/pull/1229

**PR Checklist:**
- [ ] Does the code follow accessibility standards (WCAG 2.1 AA Compliant)?
- [ ] Are all languages supported (Internationalization/Localization)?
- [ ] Are appropriate Typescript types implemented?

### CODE REVIEW

File: lib/actions/field-trip.js [hunk 1]
```diff
@@ -520,6 +523,16 @@ function prepareQueryParams(request, outbound) {
         OTP_API_TIME_FORMAT
       )
     }
+    // Generate banned list from clear list
+    if (fieldTripOptions?.allowedAgencies) {
+      const bannedAgencies = transitOperators
+        .map((to) => to.agencyId)
+        .filter((to) => !fieldTripOptions.allowedAgencies.includes(to))
```
Comment (daniel-heppner-ibigroup) [inline lib/actions/field-trip.js:530]: If you `filter` first, the second loop will have to loop over fewer items.

**Review by daniel-heppner-ibigroup — APPROVED** [review]
LGTM, one small performance suggestion

**Comment by daniel-heppner-ibigroup** [comment]
Don't forget to update otp ui

File: lib/actions/field-trip.js (hunk 1 above)
Comment (miles-grant-ibigroup) [reply lib/actions/field-trip.js:530]: You are right, but cleaner code is worth the tradeoff for this tiny list

**Review by binh-dam-ibigroup — APPROVED** [review]

**Pull request merged**